### DISCUSSION TITLE
Highlight on hover and select on click

### DISCUSSION THREAD

**Discussion by DKGH** [donalffons/opencascade.js#169] [Q&A]
Is there a way I can highlight an TopoDS shape object by hovering over it and then select it by clicking on it.

Looking for a functionality similar to what AIS interactive context provides.

**Comment by donalffons** [comment]
Currently, you would have to implement that yourself, using a JS 3D engine like ThreeJS.

In the future, I want to make it possible to use OpenCascade's / OCCT's visualization and GUI API's, which could make a use case like you're describing much simpler. I have made some experiments and a demo for a STEP file viewer, where you can see that in action: https://occt-viewer.web.app/ However, the approach I used in that application is extremely hacky and has severe pitfalls. Doing this properly would make it necessary to handle multi-threading in an improved way, which is something I want to address in an upcoming version.
![image](https://user-images.githubusercontent.com/33040347/202386412-5fa6b047-8d54-4076-8781-ca0e4075f07b.png)
